junit5/src/main/java/sunstone/core/AzureUtils.java

```diff
@@ -0,0 +1,88 @@
+import org.wildfly.extras.sunstone.api.impl.ObjectType;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.util.Set;
+
+public class AzureUtils {
+    private static ObjectProperties objectProperties = new ObjectProperties(ObjectType.JUNIT5, null);
+
+    public static AzureResourceManager getResourceManager() {
+        return AzureResourceManager
+                .authenticate(getCredentials(), new AzureProfile(AzureEnvironment.AZURE))
+                .withSubscription(objectProperties.getProperty(JUnit5Config.JUnit5.Azure.SUBSCRIPTION_ID));
+    }
+
+    private static TokenCredential getCredentials() {
+        return new ClientSecretCredentialBuilder()
+                .tenantId(objectProperties.getProperty(JUnit5Config.JUnit5.Azure.TENANT_ID))
+                .clientId(objectProperties.getProperty(JUnit5Config.JUnit5.Azure.APPLICATION_ID))
+                .clientSecret(objectProperties.getProperty(JUnit5Config.JUnit5.Azure.PASSWORD))
+                .build();
+    }
+
+    static VirtualMachine findAzureVM(AzureResourceManager arm, String name, Set<String> resourceGroups) {
+        for (String rg : resourceGroups) {
+            VirtualMachine vm = arm.virtualMachines().getByResourceGroup(rg, name);
+            if (vm != null) {
+                return vm;
+            }
+        }
+        return null;
+    }
+
+    static WebApp findAzureWebApp(AzureResourceManager arm, String name, Set<String> resourceGroups) {
+        for (String rg : resourceGroups) {
+            WebApp app = arm.webApps().getByResourceGroup(rg, name);
+            if (app != null) {
+                return app;
+            }
+        }
+        return null;
+    }
+
+    public static void waitForWebApp(WebApp app) throws InterruptedException, IOException {
+        OkHttpClient client = new OkHttpClient();
+        // todo factor
+        long timeout = 20 * 60 * 1000;
+        long start = System.currentTimeMillis();
+        while (System.currentTimeMillis() - start < timeout) {
+            try {
+                Response response = client.newCall(new Request.Builder()
+                                .url("http://" + app.defaultHostname())
+                                .method("GET", null)
+                                .build())
+                        .execute();
+                int code = response.code();
+                if (code == 404 && response.body() != null
+                        && response.body().string().contains("404 Web Site not found.")
+                        && response.body().string().contains("Custom domain has not been configured inside Azure. See <a href=\"https://go.microsoft.com/fwlink/?linkid=2194614\">how to map an existing domain</a> to resolve this.")
+                        && response.body().string().contains("Client cache is still pointing the domain to old IP address. Clear the cache by running the command <i>ipconfig/flushdns.</i>")
```

Comment:
Shouldn't these be `||`?

Reply:
No. This is catching specific 404 with all those strings. I can get that due to Azure having some issues with web app (usualy solved over time). I will add proper comment.